### ISSUE TITLE
Add constructor that accepts a signum and byte[]

### ISSUE THREAD

**Issue by footloosejava** [bwakell/Huldra#8]
BigInteger bi = new BigInteger(1,bytes)

This is essential in cryptography functions.

Can be added?

**Comment by bwakell** [comment]
Sure. Will try to get to it this weekend.

**Comment by footloosejava** [comment]
No rush bit that would be great.
Also BigInteger.mod when you get a chance.

**Comment by bwakell** [comment]
Added the constructor and a mod operation.

In the process I also stumbled upon a bug in udiv(int) and urem(int). It all (once again) came down to division behaving differently for negative longs (i.e. when the highest bit is set), a case that had previously been disregarded. For the urem(int) case it was however possible to avoid the normal procedure of dealing with unsigned long division (which includes an if-statement), and do it quite efficiently (or at least I hope so) by precomputing the remainder for 2^63 and use it when the numerator has the highest bit set (is negative), which can be done without an if-statement by exploiting signed right shift and bitwise-and. :)

The mod() and rem() operations can be further improved by writing a custom helper function which doesn't take an auxiliary array to store the quotient, but I'll leave that for another day.
